docs/hackathons/ihp_10_2022.rst

```diff
@@ -0,0 +1,276 @@
+
+
+Mathematically, this means that ``space.equip_with_metric`` transforms our smooth manifold :math:`M` into the tuple :math:`(M, g)`. The boolean ``equip`` allows to have the possibility to instantiate a non-equipped manifold. In order to keep the current behavior and for practical reasons (e.g. it can be cumbersome to equip a space that only has one "valid" metric; it is nice to have embedding spaces already equipped), ``equip=True`` by default, which means the following code is equivalent to the above:
+
+
+.. code-block:: python
+
+    space = Hypersphere(dim=2)
+
+
+We will now ask contributors to add ``_default_metric`` method to their classes, which returns the default metric for a given class (not instantiated). This will be used to equip the manifold when ``equip=True``.
+
+
+With this (seemingly small) change, we can now have (benign) circularity in our design, i.e. the space "knows" about the metric, and the metric "knows" about the space. This is achieved by requiring the first input argument of a metric to be the space:
+
+
+.. code-block:: python
+
+    class HypersphereMetric(RiemannianMetric):
+        def __init__(self, dim):
+            # init operations
+
+will become:
+
+
+.. code-block:: python
+
+    class HypersphereMetric(RiemannianMetric):
+        def __init__(self, space):
+            self._space = space
+            # other init operations
+
+
+Notice that now ``HypersphereMetric`` has access to ``dim`` via ``self._space``, which means there's only one ``dim`` (the one from ``space``) and no inconsistencies are allowed by design (e.g. equip a space with a metric that has an inappropriate ``dim``). This also removes the toll of checking input arguments for consistency.
+
+
+The ``space`` should be stored privately in the metric to avoid the recursion ``space.metric.space.metric.<...>``. The recursion is still allowed, but a user that accesses a private method from "outside" knows he/she's making a mistake.  
+
+
+Additionally, a difficult design problem is solved: how to access methods of the space within the metric? This is particularly relevant for quotient spaces (e.g. ``QuotientMetric`` requires ``FiberBundle``, and ``GraphSpaceMetric`` needs ``GraphSpace``) and was creating a design divergence where some metrics required the space, whereas others no: now all the metrics require the space.
+
+
+Moreover, we force each space to contain only one metric, since an equipped instance of a space represents :math:`(M, g)`.
+
+
+
+Similar actions will be done regarding equipping a space with a group action.
+
+
+
+Intrinsic vs extrinsic coordinates (manifold)
+*********************************************
+
+At a lower level, a manifold is expected to implement points via ``extrinsic`` OR ``intrinsic`` coordinates, never both.   
```

Comment:
I think we can distinguish two aspects:
- the coords type: extrinsic or intrinsic
- the coords: a manifold can accept several coordinate systems that are intrinsic, or several coordinate systems that extrinsic.

With this change:
- a manifold = a manifold implemented in a static coordinate system, which is additionally "tagged" with a flag "intrinsic" or "extrinsic"

Reply:
Sounds good! Do you mind to change the text to reflect this idea?